### ISSUE TITLE
Restic backup/restore get killed on a 20GB volume

### ISSUE THREAD

**Issue by Nicky31** [vmware-tanzu/velero#5903]
**What steps did you take and what happened:**
I have a nightly backup scheduled to snapshot the entire cluster including some light volumes, the largest one being a 20GB loki volume.  
Restic succeeds on others volumes, but regularly get killed on loki. Backup logs only shows the following error : 
```
pod volume backup failed: running backup, stderr=error running restic backup command restic backup ... with error: signal: killed stderr
```

Yesterday, I could get Loki snapshot this volume once by increasing memory limit from 1024 to 20248Mi, however it did not work out for this night schedule.  
Same issue on this same volume's restoration, with lower memory usage though. Might velero kill restic process for whatever reason ?

Other weird things I noticed :  
- Before I increase this memory limit, the restic pod memory did not exceed 700Mi and Kubernetes pod did not restart/get OOMKilled at all
- Now that memory limit is 2048Mi, the actual pod usage did not exceed 1400Mi. Again, no pod restart / OOM from Kubernetes perspective

Here are the restic pod (the one on loki's node & with the high memory usage ) logs :  https://gist.github.com/Nicky31/8574c987d6ff061d20aa5f291425ff0f
Backup describe :  
```
Name:         [1mvelero-fullbackup-nightly-20230222020044[0m
Namespace:    velero
Labels:       app.kubernetes.io/instance=velero
              app.kubernetes.io/managed-by=Helm
              app.kubernetes.io/name=velero
              helm.sh/chart=velero-3.1.2
              velero.io/schedule-name=velero-fullbackup-nightly
              velero.io/storage-location=default
Annotations:  helm.sh/hook=post-install,post-upgrade,post-rollback
              helm.sh/hook-delete-policy=before-hook-creation
              velero.io/source-cluster-k8s-gitversion=v1.21.7-r0-CCE21.11.1.B007
              velero.io/source-cluster-k8s-major-version=1
              velero.io/source-cluster-k8s-minor-version=21+

Phase:  [31mPartiallyFailed[0m (run `velero backup logs velero-fullbackup-nightly-20230222020044` for more information)

Errors:    1
Warnings:  0

Namespaces:
  Included:  *
  Excluded:  velero

Resources:
  Included:        *
  Excluded:        endpointslices.discovery.k8s.io
  Cluster-scoped:  auto

Label selector:  <none>

Storage Location:  default

Velero-Native Snapshot PVs:  auto

TTL:  240h0m0s

CSISnapshotTimeout:  10m0s

Hooks:  <none>

Backup Format Version:  1.1.0

Started:    2023-02-22 03:00:44 +0100 CET
Completed:  2023-02-22 03:04:54 +0100 CET

Expiration:  2023-03-04 03:00:44 +0100 CET

Total items to be backed up:  1293
Items backed up:              1293

Resource List:
...

Velero-Native Snapshots: <none included>

restic Backups:
  Completed:
    ...
  Failed:
    core/core-loki-0: storage
```

Side questions :  
- Is it normal for restic to use "that much" memory for just 20GB ? Or is it just designed to use all free memory up ? 
- Why the restic pod logs have some "PodVolumeBackup starting" from 9AM although no schedule / manual command exists at that time ?

**What did you expect to happen:**
The loki volume snapshot to succeeds

**Environment:**

- Velero version (use `velero version`): v1.10.1
- Velero features (use `velero client config get features`):  NOT SET
- Kubernetes version (use `kubectl version`): Client 1.21 & server on 1.19+
- Kubernetes installer & version: Cloud Container Engine
- Cloud provider or hardware configuration:  OBS
- OS (e.g. from `/etc/os-release`): EulerOS 2.0

**Vote on this issue!**

This is an invitation to the Velero community to vote on issues, you can see the project's [top voted issues listed here](https://github.com/vmware-tanzu/velero/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc).  
Use the "reaction smiley face" up to the right of this comment to vote.

- :+1: for "I would like to see this bug fixed as soon as possible"
- :-1: for "There are more important bugs to focus on right now"

**Comment by qiuming-best** [comment]
For the Restic process killed Event, there are two reasons for it. 
One it's OOM, you could increase memory.
Another one is a timeout for the backup/restore.  you could increase the default timeout by the flag `fs-backup-timeout` when creating backup.
through the logs you provided, Restic is been killed by timeout.
Also, you can try using Kopia uploader to make your backup more efficient.

**Comment by Nicky31** [comment]
From the backup description it would have finished only 4 minutes after starting 🧐 However I ran another manual cluster backup this morning and I confirm that for whatever reason loki's volume backup get stuck at 99.64% for almost 1h hour.  
I had a similar bug on another volume backup which went quite easily up to 100% & got stuck InProgress. at 100% for a while.

Indeed I've made a fews test with Kopia and performances look overall better (& no stuck issue), I think we'll stick with it, thanks ! 
With Kopia the first loki backup took about 25min, and restoring this took almost 1h : both numbers look a bit slow to me, but still working better than restic. 

I guess we can close this
